source/common/http/date_provider_impl.cc

```diff
@@ -6,7 +6,12 @@
 namespace Envoy {
 namespace Http {
 
-DateFormatter DateProviderImplBase::date_formatter_("%a, %d %b %Y %H:%M:%S GMT");
+class TlsCachingDateFormatter : public DateFormatter {
```

Comment:
What is TLS specific about this class? (I'm guessing that means Thread Local Storage not Transport Level Security, but I don't think I see what is thread local about this?

nit: Maybe this should be in an anon namespace?